nco/custom.py

```diff
@@ -0,0 +1,268 @@
+    #def __init__(self, **kwargs):
+        mode = kwargs.pop('mode', mode)
+        attName = kwargs.pop('attName', attName)
+        varName = kwargs.pop('varName', varName)
+        Value = kwargs.pop('Value', Value)
+        sType = kwargs.pop('sType',sType )
+            
+        if mode in validModes:
+           mode=validModes[mode]  
+        elif mode in validModes.keys():
+           pass
+        else:
+           raise Exception('{0}: mode "{1}" not found'.format("atted()","atted():",mode))
+        
+        if attName==None or len(attName)==0: 
+            raise Exception ('{0}: attName zero length\n'.format("atted()"))
+
+
+        # set member defaults 
+        self.mode=mode
+        self.attName=attName
+        self.varName=varName 
+        self.npType=None
+        self.npValue=None    
+
+        # dont bother about type & value
+        if self.mode=="d":
+            return None
+        
+        # deal with string quirk 
+        # if user specifies type 'string' or 'sng' with a single character string 
+        # the the output type should be 'sng and NOT 'c'. To do this we put the single string
+        # inside a list the the prnOptions() method treats it as it would a list of strings
+
+        if type(Value) is str and ( sType == 'string' or sType == 'sng'):  
+          Value=[Value]
+
+        # see if Value is iterable   
+        try:
+            if type(Value) is str:
+              raise TypeError("str is not iterable (for our purposes")
+            it=iter(Value)    
+            inputType=type(next(it))   
+            bIterable=True
+            del it   
+        except Exception:
+            bIterable=False 
+            inputType=type(Value) 
+         
+         
+        if sType:
+            try:    
+                npType=npTypesNP[sType] 
+            except:
+                raise Exception('{0}: specified Type "{1}" not found.\nValid values {2}\n'.format("atted()", sType, npTypes.keys()))                          
+        # set default types
+        else: 
+           if inputType is int:
+               npType=np.int32
+           elif inputType is float:
+               npType=np.float64
+           elif inputType is str:
+               npType=str
+           # check the input type 
+           else:  
+               try: 
+                   x=npTypes[str(np.dtype(inputType)) ] 
+                   npType=inputType
+               except:
+                   raise Exception('{0}: The type of Value "{1}" is NOT valid\nValid values {2}\n'.format("atted()", inputType, npTypes.keys()))                          
+
+        if bIterable:              
+            if npType is str:                          
+                # convert everything to string
+                sList=[ np.dtype(str).type(v) for v in Value]
+               
+                # get max string length
+                lenMax=len ( max(sList,key=len) )
```

Comment:
pep8: take a look at where spaces should (not) be.

https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements